tests/common.py

```diff
@@ -1260,7 +1260,14 @@ async def mock_write_data(
         # To ensure that the data can be serialized
         _LOGGER.debug("Writing data to %s: %s", store.key, data_to_write)
         raise_contains_mocks(data_to_write)
-        data[store.key] = json.loads(json.dumps(data_to_write, cls=store._encoder))
+        encoder = store._encoder
+        if encoder and encoder is not JSONEncoder:
+            # If they pass a custom encoder that is not the
+            # default JSONEncoder, we use the slow path of json.dumps
+            dump = ft.partial(json.dumps, cls=store._encoder)
+        else:
+            dump = _orjson_default_encoder
+        data[store.key] = json.loads(dump(data_to_write))
 
```

Comment:
I switched the test to behave like the production code since the production code was being mocked out here and I missed this bit of mocking when I did the conversion.

Once I did that there were some failing tests were datetime was subclassed